day2

ShayRasooly opened the pull request.

EvgeniaKantor committed:
daily challenge

EvgeniaKantor committed:
daily challenge

ShayRasooly commented on Week1/Day2/0304_Daily Challenge.py:
```diff
@@ -0,0 +1,16 @@
+first_number = int(input("Enter the first number: "))
+length = int(input("Enter the length of the list: "))
+
+my_list = [first_number * (i+1) for i in range(length)]
```
nice list comprehension. is there a way to get rid of this +1?

EvgeniaKantor replied:
I really don't know, so I had to use +1

ShayRasooly replied:
you can put this +1 inside the range

ShayRasooly commented on Week1/Day2/0304_Daily Challenge.py:
```diff
@@ -0,0 +1,16 @@
+new_word = ''
+previous_char = ''
+for char in user_word:
+    if char != previous_char:
```
efficient way of thinking, maybe make more clear for future debuggers

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+my_fav_numbers.remove(100)
+
+friend_fav_numbers = set({6, 4})
+our_fav_numbers = my_fav_numbers.union(friend_fav_numbers)
```
nice union

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+  "salary": 8000,
+  "city": "New york"
+}
+keys_to_remove = set({"name", "salary"})
```
why set?

EvgeniaKantor replied:
I must have tried something and forgot to delete the exercise.

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+}
+keys_to_remove = set({"name", "salary"})
+for key in keys_to_remove:
+    sample_dict.pop(key, None)
```
why None?

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+print(basket)
+basket.append("Kiwi")
+print(basket)
+basket.append("Apples")
```
wrong. how do you add an item to the beginning of the list?

EvgeniaKantor replied:
Thank you! Fixed it.

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+
+#exercise4:
+#An integer (more commonly called an int) is a number without a decimal point. A float is a floating-point number, which means it is a number that has a decimal place.
+first = float(1.5)
```
why the conversion to float? what 1.5 was before?

EvgeniaKantor replied:
for sure))) i changed

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+for number in numbers:
+    print(number)
+for number in numbers:
+    if number % 2 == 0:
```
what is the difference between this number and its index?

EvgeniaKantor replied:
Ok. I've changed with enumerate

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+    user_name = input("Please enter your name: ")
+    if user_name == my_name:
+        print("It's true!")
+        break
```
can you avoid using break?

EvgeniaKantor replied:
This loop is working, but I added a new code without break

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+    ask_fruit = input('Please enter one fruit: ')
+    if ask_fruit in favorite_fruits:
+        print("Enjoy your favorite fruit!")
+        break
```
why the break?

EvgeniaKantor replied:
This loop is working, but I added a new code without break

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+    else:
+        print("You chose a new fruit. I hope you enjoy!")
+#exercise8:
+tops = [1,2,3,4]
```
why bother using numbers?

EvgeniaKantor replied:
because I first wanted to use a dictionary, but I couldn't, but the numbers stayed.))

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+price = float(10)
+toppings_list = []
+while True:
+    ask = input('Please enter the number of extra tops for pizza using this list: Cheese 1, Veggies 2, Sauces 3, Pesto 4 or type done:')
```
maybe better var name than "ask"?

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+    else:
+        print('Invalid')
+your_price = price + 2.5 * len(toppings_list)
+print("total price is: ", your_price)
```
what about letting him know what will be on his pizza?

EvgeniaKantor replied:
I changed

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+
+sandwich_orders = ["Tuna sandwich", "Pastrami sandwich", "Avocado sandwich", "Pastrami sandwich", "Egg sandwich", "Chicken sandwich", "Pastrami sandwich"]
+while "Pastrami sandwich" in sandwich_orders:
+    sandwich_orders.remove("Pastrami sandwich")
```
why twice?

EvgeniaKantor replied:
just mistake, I fixed it

ShayRasooly commented on Week1/Day2/Exercise XP.py:
```diff
@@ -0,0 +1,142 @@
+print(sandwich_orders)
+finished_sandwiches = []
+for sandwiches in sandwich_orders:
+    finished_sandwiches.append(sandwiches) and sandwich_orders.remove(sandwiches)
```
not recommended to manipulate the list while iterating over it

EvgeniaKantor replied:
I've changed

EvgeniaKantor committed:
daily challenge

EvgeniaKantor committed:
daily challenge

EvgeniaKantor committed:
new

EvgeniaKantor committed:
new

EvgeniaKantor committed:
new

EvgeniaKantor committed:
new

EvgeniaKantor committed:
new

EvgeniaKantor committed:
Week2_0310

EvgeniaKantor committed:
Week2_0310

EvgeniaKantor committed:
Week2_day3

EvgeniaKantor committed:
Week2_day4

EvgeniaKantor committed:
Week2_day5

EvgeniaKantor committed:
Week3_day1

EvgeniaKantor committed:
Week3_day1

EvgeniaKantor committed:
w3d1

EvgeniaKantor committed:
w3d2

EvgeniaKantor committed:
Week3_day3

EvgeniaKantor committed:
Merge remote-tracking branch 'DI-Bootcamp/master'

EvgeniaKantor committed:
w3d3

EvgeniaKantor committed:
Week3_day4

EvgeniaKantor committed:
Merge remote-tracking branch 'DI-Bootcamp/master'

EvgeniaKantor committed:
hackathon_21.03_16:30

EvgeniaKantor committed:
hackathon_21.03_19:15

EvgeniaKantor committed:
hackathon_21.03_19:15

EvgeniaKantor committed:
hackathon_21.03_19:15

EvgeniaKantor committed:
hackathon_21.03_20:05

EvgeniaKantor committed:
hackathon_22.03_15:12

EvgeniaKantor committed:
hackathon_22.03_15:12

EvgeniaKantor committed:
hackathon_22.03_15:12

EvgeniaKantor committed:
hackathon_22.03_15:12

EvgeniaKantor committed:
hackathon_22.03_15:12

EvgeniaKantor committed:
hackathon_22.03_15:12

EvgeniaKantor committed:
hackathon_22.03_15:12

EvgeniaKantor committed:
Week3day4

EvgeniaKantor committed:
hackathon_23.03_22:40

EvgeniaKantor committed:
hackathon_23.03_22:40

EvgeniaKantor committed:
hackathon_23.03_22:40

EvgeniaKantor committed:
test

EvgeniaKantor committed:
Daily challenge (**Step 1: Search for a relevant case or news story.**

Title: "How Data Analytics Is Revolutionizing Cancer Treatment"

Source: Scientific American

Summary: The article discusses how data analytics is transforming cancer treatment by enabling personalized medicine approaches and improving patient outcomes.

**Step 2: Analyze the role of data analysis in the chosen story.**

In this case, data analysis plays a pivotal role in several aspects of cancer treatment:

- **Genomic Data Analysis:** Researchers analyze genomic data from cancer patients to identify genetic mutations associated with different types of cancer. This involves sequencing the DNA of tumor samples and comparing them to healthy tissue samples to pinpoint genetic abnormalities.

- **Machine Learning Algorithms:** Data analysis techniques such as machine learning are used to analyze large datasets of patient information, including genomic data, medical imaging, treatment history, and outcomes. These algorithms can identify patterns and correlations that may not be apparent to human researchers, helping to predict how individual patients will respond to different treatments.

- **Treatment Personalization:** By analyzing data on genetic mutations and treatment outcomes across a large patient population, oncologists can tailor treatment plans to individual patients. For example, if a certain genetic mutation is associated with resistance to a particular chemotherapy drug, the patient may be prescribed a different treatment regimen that is more likely to be effective.

- **Clinical Decision Support Systems:** Data analysis tools are integrated into clinical decision support systems that help oncologists make informed treatment decisions. These systems provide evidence-based recommendations for treatment options based on the patient's specific genetic profile and medical history.

**Step 3: Discuss within your group how the data analysis impacted the situation.**

- **Without Data Analysis:** Without data analysis, oncologists would have limited information to guide treatment decisions. They would rely primarily on empirical evidence from clinical trials and general treatment guidelines rather than personalized insights based on individual patient data.

- **Contribution to Solving Problems:** Data analysis enables oncologists to identify genetic mutations driving cancer growth and resistance to treatment, allowing for more targeted and effective therapies. By analyzing large datasets of patient information, researchers can also discover new biomarkers and therapeutic targets, leading to innovations in cancer treatment.

- **Impact on Patient Outcomes:** Data analysis has a significant impact on patient outcomes by facilitating personalized treatment approaches. Patients receive therapies that are more likely to be effective based on their unique genetic profiles, resulting in improved survival rates and quality of life.

In summary, data analysis plays a crucial role in revolutionizing cancer treatment by enabling personalized medicine approaches, improving treatment outcomes, and advancing our understanding of the molecular mechanisms driving cancer progression)

EvgeniaKantor committed:
Daily challenge (I mistakenly switched places with the exercises)

EvgeniaKantor committed:
Hackathon2

EvgeniaKantor committed:
Week7 Day1

EvgeniaKantor committed:
Week7 Day1

EvgeniaKantor committed:
Week7 Day2

gavishap committed:
Create feedback.md

EvgeniaKantor committed:
Create Daily_Challenge_W3D3

EvgeniaKantor committed:
Delete Week3/Day3/Daily_Challenge_W3D3

EvgeniaKantor committed:
Create Daily_Challenge_W3D3.sql

EvgeniaKantor committed:
Week7 Day3

EvgeniaKantor committed:
Merge remote-tracking branch 'DI-Bootcamp/master'

EvgeniaKantor committed:
Hackathon 3

EvgeniaKantor committed:
week 10

EvgeniaKantor committed:
Add files via upload

Week 10 Day 2

EvgeniaKantor committed:
Add files via upload

Week 10 Day 2 Ex 2

EvgeniaKantor committed:
Add files via upload

EvgeniaKantor committed:
W10 D2 Ex4

EvgeniaKantor committed:
W10 D2 EX5

EvgeniaKantor committed:
W10 D2 Daily Challenge

gavishap committed:
Create feedback.md

EvgeniaKantor committed:
Add files via upload

Week 10 Mini project

EvgeniaKantor committed:
week 11

EvgeniaKantor committed:
Merge remote-tracking branch 'DI-Bootcamp/master'

EvgeniaKantor committed:
Add files via upload

Week 11 D 1

EvgeniaKantor committed:
Add files via upload

Week 11 Day 03

EvgeniaKantor committed:
Add files via upload

EvgeniaKantor committed:
Add files via upload

Week 11 D02 (in Tableau)

EvgeniaKantor committed:
Week 10(11) Day 4

EvgeniaKantor committed:
Delete Week11/Daily_Challenge_W11D4.twb

EvgeniaKantor committed:
Week 10 (11) Day 4

EvgeniaKantor committed:
final project

EvgeniaKantor committed:
Merge remote-tracking branch 'DI-Bootcamp/master'
